radis/io/exomol.py

```diff
@@ -8,14 +8,125 @@
 import pathlib
 
 import numpy as np
+import pandas as pd
 
 from radis.api.exomolapi import (
     MdbExomol,
     get_exomol_database_list,
     get_exomol_full_isotope_name,
+    read_states,
 )
 from radis.db.classes import get_molecule_identifier
 
+# ExoMol state/transition processing and degeneracy calculation logic
+def _map_states_to_transitions(df, states_df):
+    """Map electronic state information from energy levels to transitions."""
+    if 'Es' in states_df.columns:
+        state_map = dict(zip(states_df['i'], states_df['Es']))
+    elif 'state' in states_df.columns:
+        state_map = dict(zip(states_df['i'], states_df['state']))
+    else:
+        print("Warning: No electronic state column found in states DataFrame")
+        return df
+    df['state_lower'] = df['i_lower'].map(state_map)
+    df['state_upper'] = df['i_upper'].map(state_map)
+    return df
+
+def _get_electronic_state_energy(state_label):
+    """Get electronic state energy from ExoMol state label."""
+    if state_label is None:
+        return 0.0
+    OH_STATE_ENERGIES = {
+        'X2Pi': 0.0,
+        'A2Sigma+': 32600.0,
+        'B2Sigma+': 50000.0,
+        'C2Sigma+': 75000.0,
+    }
+    base_state = state_label.split()[0].split('-')[0]
+    if base_state in OH_STATE_ENERGIES:
+        return OH_STATE_ENERGIES[base_state]
+    else:
+        raise ValueError(f"Unknown electronic state: {base_state}. Known states: {list(OH_STATE_ENERGIES.keys())}")
+
+def _convert_branch_format(branch_str):
+    """Convert ExoMol branch format to RADIS numeric format."""
+    if pd.isna(branch_str):
+        return 0
+    try:
+        num = int(branch_str.split('/')[0])
+        if num == 1:
+            return 1
+        elif num == 3:
+            return -1
+        else:
+            return 0
+    except (ValueError, IndexError):
+        return 0
```

Comment:
Return more explicit error message please. otherwise, there is no need to put a "try"